### ISSUE TITLE
Build error when compiling on Ubuntu 16

### ISSUE THREAD

**Issue by Thread7** [RavenProject/Ravencoin#14]
I followed the instructions on building for Ubuntu 16.  i didn't get any errors installing the required packages.  
sudo ./autogen worked fine
sudo ./configure worked fine
But after I issued "make" I get the error:
**/usr/bin/ld: crypto/libraven_crypto.a(crypto_libraven_crypto_a-blake.o): relocation R_X86_64_32 against `.rodata' can not be used when making a shared object; recompile with -fPIC**

`Making all in src
make[1]: Entering directory '/home/myuser/Ravencoin/src'

make[2]: Entering directory '/home/myuser/Ravencoin/src'
make[3]: Entering directory '/home/myuser/Ravencoin'
make[3]: Leaving directory '/home/myuser/Ravencoin'
make[3]: Entering directory '/home/myuser/Ravencoin/src/secp256k1'
make[3]: Leaving directory '/home/myuser/Ravencoin/src/secp256k1'
  CXXLD    ravend
/usr/bin/ld: crypto/libraven_crypto.a(crypto_libraven_crypto_a-blake.o): relocation R_X86_64_32 against '.rodata' can not be used when making a shared object; recompile with -fPIC
crypto/libraven_crypto.a: error adding symbols: Bad value
collect2: error: ld returned 1 exit status
Makefile:3835: recipe for target 'ravend' failed
make[2]: *** [ravend] Error 1
make[2]: Leaving directory '/home/myuser/Ravencoin/src'
Makefile:9947: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/myuser/Ravencoin/src'
Makefile:747: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1`

I'm guessing I should rerun ./configure with some sort of flag like this:
sudo ./configure CXXFLAGS=-fPIC
But that exact statement doesn't work because I get the exact same error when I try make.
Please advise.

**Comment by todd1251** [comment]
I use this on Ubuntu 16.04:

```
sudo apt-get update && sudo apt-get install -y software-properties-common
sudo add-apt-repository -y ppa:bitcoin/bitcoin
sudo apt-get update && sudo apt-get install -y automake autotools-dev bsdmainutils build-essential git libboost-all-dev libdb4.8-dev libdb4.8++-dev libevent-dev libminiupnpc-dev libprotobuf-dev libqrencode-dev libqt5gui5 libqt5core5a libqt5dbus5 libssl-dev libtool libzmq3-dev pkg-config protobuf-compiler python3 qttools5-dev qttools5-dev-tools
git clone https://github.com/RavenProject/Ravencoin.git
cd Ravencoin
./autogen.sh
./configure --enable-cxx --disable-shared --with-pic --prefix=$BDB_PREFIX CXXFLAGS="-fPIC" CPPFLAGS="-fPIC"
make
cd src
cp ravend raven-tx raven-cli qt/raven-qt ~
```

**Comment by Thread7** [comment]
This worked.  I had to do "make clean" first.  I'm going to leave this issue open for just a bit since I've got to believe others will have this problem.  I had tried the -fPIC variable but everything else that todd1251 wrote above I would have had no idea which ones to use.

**Comment by VERTlG0** [comment]
Thanks for sharing! I was scratching my head with what else I had to do.. (installed openssl and automake-1.14)

**Comment by Jp1337** [comment]
Hello,
I'm having this exact same issue with Fedora 26. Also tried only with the CXXFLAGS="-fPIC" option, but as you pointed out it made no difference.
Just as a suggestion to devs, maybe put some additional hints for building into the readme
Thanks!

**Comment by spyder46n2** [comment]
Fixed in #110
